### PR TITLE
CI test framework

### PR DESCRIPTION
**Pull request by tom-rosewell**
Infra is the same as python, everything that's tested is a C# script

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,31 +1,19 @@
...
-ENV PRE_RELEASE_VERSION 3.2.8-pre
-ENV PRE_RELEASE_BUILD r5705
-ENV PRE_RELEASE_SOURCE http://sdk.jenkins.couchbase.com/job/dotnet/job/sdk/job/couchbase-net-client-scripted-build-pipeline/lastSuccessfulBuild/artifact/couchbase-net-client-${PRE_RELEASE_VERSION}-${PRE_RELEASE_BUILD}.zip
+COPY . /dotnet-docs-repo
+WORKDIR /dotnet-docs-repo/tests
 
-RUN mkdir -p /app/nuget-sources/
-WORKDIR /app/nuget-sources/
-RUN curl -O ${PRE_RELEASE_SOURCE}
-RUN dotnet nuget add source /app/nuget-sources/
```
Comment (osfameron) [inline Dockerfile:18]: we don't add the SDK at all?
Ah, because it's all specified in the .csx nuget comments?

Comment (tom-rosewell) [reply Dockerfile:18]: Yeah

File: tests/wait-for-couchbase.sh [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+# install beer-sample
+curl -u Administrator:password -X POST http://${CB_HOST}:8091/sampleBuckets/install -d '["beer-sample"]'
+
+# Give it a bit to stablise
+sleep 3
+log Couchbase Server ready
+
+# Check if we should run the tests
+if [ $1 -eq 1 ]
```
Comment (osfameron) [inline tests/wait-for-couchbase.sh:114]: this is a horrible magic constant.
Didn't wait-for-couchbase already exec a command passed in?
Then we could just set the ENTRYPOINT to pass `bats test.bats` instead?

File: docker-compose.yml [hunk 3]
```diff
@@ -1,27 +1,41 @@
...
-    volumes:
-      - .:/app/docs-sdk-dotnet
-
-  db:
-    image: build-docker.couchbase.com:443/couchbase/server-internal:7.1.0-2549
+  # Docker uses services names as DNS records.
+  # Calling the db container this means the
+  # connection strings in the samples work but,
+  # still look like placeholders.
+  your-ip:
```
Comment (osfameron) [inline docker-compose.yml:8]: oh, I see. cute.

File: tests/wait-for-couchbase.sh (hunk 2 above)
Comment (tom-rosewell) [reply tests/wait-for-couchbase.sh:114]: It did, if I remember correctly there was a reason it didn't work when I was making python. Will go do some investigating

File: docker-compose.yml [hunk 4]
```diff
@@ -1,27 +1,41 @@
...
+    container_name: dotnet-sdk
+    profiles: 
+      - prod
+
+  # This profile is for local use creating tests.
+  local-dotnet-sdk-repo:
+    build:
+      dockerfile: local-tests.Dockerfile
+    depends_on:
+      - your-ip
```
Comment (osfameron) [inline docker-compose.yml:36]: this looks a bit odd. maybe `your-couchbase-server` might be a clearer placeholder?
Meh, nah, it's fine, `your-ip` is probably clearer elsewhere.

**Review by osfameron — APPROVED** [review]
Looks good 👍 
Good work Tom!

(Some niggle comments about e.g. the magic constant to run tests, which I'm not super happy about, but it can always be tidied away later if we find a good pattern.)

File: tests/wait-for-couchbase.sh (hunk 2 above)
Comment (osfameron) [reply tests/wait-for-couchbase.sh:114]: OK. That's odd.
But hey, I don't love this solution, but it's just a niggle, can merge and come back to it in due course 👍

**Review by RichardSmedley — APPROVED** [review]
Thanks @tom-rosewell - this is great to have  :+1: 

And thanks @osfameron for the extra pair of eyes - agree with your comments, but am more relaxed about the sh script (at least as a will-improve-it-eventually type item).

**Comment by tom-rosewell** [comment]
Ah I remember why - for some reason putting a sleep infinity in the entrypoint breaks the wait for couchbase (I have no idea how this happens)

**Pull request merged**